subiquity/models/filesystem.py

```diff
@@ -1110,9 +1135,29 @@ class ZFS:
     # options to pass to zfs dataset creation
     properties: Optional[dict] = None
 
+    @property
+    def fstype(self):
+        return 'zfs'
+
+    @property
+    def canmount(self):
+        return get_canmount(self.properties, False)
```

Comment:
The previous version sure looked like it had different defaults for canmount for zfs and zpool objects. Want to check that detail? (I certainly don't know what is correct here!)

Reply:
Intentional!  I have found it helpful to think of a ZPool as implicitly creating a ZFS object, and the fs_properties on the zpool should be equivalent to the properties on a zfs.

Then I made sure the above matched curtin's default behavior, and I think we should revisit that, as curtin defaults canmount='off' while zfsutils-linux defaults to canmount='on', for both a zfs dataset and the implicit zfs dataset created by the zpool.

So for right now I think this part is correct, and we should ponder https://git.launchpad.net/curtin/tree/curtin/block/zfs.py#n19 .